.github/workflows/python.yml

```diff
@@ -0,0 +1,73 @@
+name: Python
+
+on:
+  workflow_call:
+
+jobs:
+  formatting:
+    name: Formatting
+    runs-on: ubuntu-latest
+    timeout-minutes: 2
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install poetry
+        run: |
+          pipx install poetry
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version-file: '.python-version'
+          cache: 'poetry'
+      - name: Restore cache
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: ${{ runner.os }}-venv-${{ hashFiles('poetry.lock') }}
+          restore-keys: ${{ runner.os }}-venv
+      - name: Install Dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          poetry install -E parsers
+      - name: Check Black formatting
+        run: |
+          poetry run black --check .
+      - name: Check isort formatting
+        run: |
+          poetry run isort -c .
+
+  test:
+    name: Tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    needs: [formatting]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install poetry
+        run: |
+          pipx install poetry
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version-file: '.python-version'
+          cache: 'poetry'
+      - name: Restore cache
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: ${{ runner.os }}-venv-${{ hashFiles('poetry.lock') }}
+          restore-keys: ${{ runner.os }}-venv
+      - name: Install libxml2-dev and tesseract-ocr
+        run: |
+          sudo apt-get update
+          sudo apt-get install libxml2-dev tesseract-ocr tesseract-ocr-eng
+      - name: Install Poetry dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          poetry install -E parsers
+      - name: Run tests
+        run: |
+          poetry run test
```

Comment:
We run the formating (isort and black) in a speperate job so no (easier to identify the problem at first glance that way).

Reply:
Okay I see. Can you replace the commands in the formatting steps with `poetry run lint` and maybe rename it to `linting`. That way we only need to keep one place updated. Otherwise the two files might get out-of-sync

Reply:
I can do that but then we would loose the individual steps in that job:
<img width="198" alt="image" src="https://user-images.githubusercontent.com/30777521/177590548-1959a078-f043-407e-ae6e-3aabf797ee75.png">

But I can still update the PR if you want.

Reply:
Ah, good point. Though I think that is a tradeoff that we have to make for now, unfourtunately.